rotkehlchen/api/rest.py

```diff
@@ -4062,3 +4063,74 @@ def import_user_assets(self, path: Path) -> Response:
             )
 
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
+
+    @require_loggedin_user()
+    def export_user_db_snapshot(self, timestamp: Timestamp) -> Response:
+        timed_location_data = self.rotkehlchen.data.db.get_timed_location_data(timestamp)
+        processed_timed_location_data = process_result_list(timed_location_data)
+
+        timed_balances = self.rotkehlchen.data.db.get_timed_balances(timestamp)
+        processed_time_balances_data = process_result_list(timed_balances)
+
+        tmp_dir = tempfile.TemporaryDirectory()
+        tmp_dir_path = Path(tmp_dir.name)
+        timed_balances_csv_path = tmp_dir_path / 'timed_balances.csv'
+        timed_location_data_csv_path = tmp_dir_path / 'timed_location_data.csv'
+        snapshot_zip_path = tmp_dir_path / 'snapshot.zip'
```

Comment:
This is a lot of code to be here.

And it also duplicates in the `download_db_user_snapshot()` function below.

You know what would be a good idea? If you made a new DB class under https://github.com/rotki/rotki/tree/develop/rotkehlchen/db in a new file called snapshots.py. This way you could abstract all the code there, along with the changes you put in dbhandler.py  (we try to break stuff out of that huge file, not add more :smile: ).

This way you can abstract the entire fucntionality there and call it from here and from `download_user_snapshot` with different arguments denoting the different way of getting the CSV.

This will also be helpful since we have more work to be done with the snapshots in the future, such as deleting certain snapshots or editing them.